encfs/BlockFileIO.cpp

```diff
@@ -147,6 +147,10 @@ ssize_t BlockFileIO::read(const IORequest &req) const {
       }
 
       ssize_t readSize = cacheReadOneBlock(blockReq);
+      if (readSize < 0) {
+        result = readSize;
```

Comment:
Just committed :
_Returns the number of bytes read, or -errno in case of failure._